packages/terra-functional-testing/src/commands/expect/toMatchReference.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Helper method to determine which comparison results are relevant if the screenshot assertion
+ * fails.
+ *
+ * @param {Object} screenshot - The comparison result
+ * @param {boolean} screenshot.isSameDimensions - If the latest screenshot was the same size as the reference screenshot.
+ * @param {boolean} screenshot.isWithinMismatchTolerance - If the latest screenshot was within the mismatch tolerance.
+ * @param {boolean} screenshot.isExactSameImage - If the latest screenshot matched the reference screenshot exactly, i.e. 0% mismatch.
+ * @param {Number} screenshot.misMatchPercentage - The mismatch percentage when comparing the latest screenshot to the reference screenshot.
+ * @returns {String} - A message describing the status of the screenshot.
+ */
+function getComparisonResult(screenshot) {
+  const { misMatchPercentage, isSameDimensions } = screenshot;
+  const relevantInformation = {};
+
+  if (!isSameDimensions) {
+    relevantInformation.isSameDimensions = isSameDimensions;
+  }
+
+  relevantInformation.misMatchPercentage = misMatchPercentage;
+
+  return `${JSON.stringify(relevantInformation, null, 2)}`;
+}
```

Comment:
nit: (These aren't necessary changes, just suggestions)

This function is heavy for what is essentially just formatting a string. We could make this more human friendly by formatting the expect message sentence.

Mismatch:

> expected the screenshot to be within the mismatch tolerance, but received a mismatch difference of 0.23%.

Dimension Changes

> expected the screenshot to match reference, but received a screenshot with different dimensions

Reply:
Restructured the error message to make it more human friendly to read. The `getComparisonResult` method is no longer needed.
https://github.com/cerner/terra-toolkit/pull/545/commits/9933873194e2d7e6d395853f313cdd4cf89d1212